src/main/java/bio/terra/janitor/service/janitor/JanitorService.java

```diff
@@ -22,14 +24,24 @@
 
   private final JanitorDao janitorDao;
   private final TransactionTemplate transactionTemplate;
+  private final IamService iamService;
 
   @Autowired
-  public JanitorService(JanitorDao janitorDao, TransactionTemplate transactionTemplate) {
+  public JanitorService(
+      JanitorDao janitorDao, TransactionTemplate transactionTemplate, IamService iamService) {
     this.janitorDao = janitorDao;
     this.transactionTemplate = transactionTemplate;
+    this.iamService = iamService;
   }
 
-  public CreatedResource createResource(CreateResourceRequestBody body) {
+  public CreatedResource createResource(
+      CreateResourceRequestBody body, AuthenticatedUserRequest userReq) {
+    iamService.isAdminUser(userReq);
+    return createResourceInternal(body);
+  }
+
+  /** Internal method to Create a new {@link TrackedResource} without user credentials. */
+  public CreatedResource createResourceInternal(CreateResourceRequestBody body) {
```

Comment:
nit: no longer public?

Reply:
unfortunately it need to be public because classes in other package would need this. (TrackedResourceSubscriber)